java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -181,20 +183,66 @@ private FlowComputation(Set<SymbolicValue> symbolicValues, Predicate<Constraint>
     return flows;
   }
 
-  Stream<ExecutionPath> startPath(ExplodedGraph.Edge edge, PSet<Symbol> trackedSymbols) {
-    return new ExecutionPath(null, PCollections.emptySet(), trackedSymbols, ImmutableList.of(), false).addEdge(edge);
+  Stream<ExecutionPath> startPath(ExplodedGraph.Edge edge, PSet<Symbol> trackedSymbols, SameConstraints sameConstraints) {
+    return new ExecutionPath(null, PCollections.emptySet(), trackedSymbols, sameConstraints, ImmutableList.of(), false).addEdge(edge);
+  }
+
+  private static class SameConstraints {
+    private final Map<Symbol, Boolean> sameConstraintsBySymbol;
```

Comment:
Thanks for the idea. I tried with a `PMap` and lost myself in some details when I could have simply use a `PSet`.